File: Regression_Predictor/Polynomial Regression/polynomialregression_nicholas.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Position_Salaries.csv')
x = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values
"""
#splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)
"""
#fitting linear regression to the dataset
from sklearn.linear_model import LinearRegression
lin_reg = LinearRegression()
lin_reg.fit(x, y)

#fitting polynomial regression to the dataset
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 5)
x_poly = poly_reg.fit_transform(x)
lin_reg_2 = LinearRegression()
lin_reg_2.fit(x_poly, y)

#visualising the linear regression results 
plt.scatter(x, y, color = 'red')
plt.plot(x, lin_reg.predict(x), color = 'blue')
plt.title('Truth or Bluff(linear regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()
#visualising the polynomial regression results
x_grid = np.arange(min(x), max(x), 0.01)
x_grid = x_grid.reshape(len(x_grid), 1)
plt.scatter(x, y, color = 'red')
plt.plot(x_grid, lin_reg_2.predict(poly_reg.fit_transform(x_grid)), color = 'blue')
plt.title('Truth or Bluff(polynomial regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

#predicting a new result with linear regression
lin_reg.predict(np.array(6.5).reshape(1,-1))

#predicting a new reslut with polynomial regression
lin_reg_2.predict(poly_reg.fit_transform(np.array(6.5).reshape(1,-1)))

File: Regression_Predictor/Polynomial Regression/polynomialregression_nicholas.R
#data preprocessing

#importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[ , 2:3]
#dataset = dataset[, 2:3]
#splitting the dataset into the Train set and Test set
# library(caTools)
# set.seed(123)
# split = sample.split(dataset$Purchased, SplitRatio = 0.8)
# training_set = subset(dataset, split == TRUE)
# test_set = subset(dataset, split == FALSE)

#Fitting linear regression to the dataset
lin_reg = lm(formula = Salary~. , 
             data = dataset)

#Fitting polynomial regression to the dataset
dataset$level2 = dataset$Level^2
dataset$level3 = dataset$Level^3
dataset$level4 = dataset$Level^4
poly_reg = lm(formula = Salary~. , 
             data = dataset)

#visualising the linear regression results(blue)
#visualising the linear regression results(green)
library(ggplot2)
ggplot()+
  geom_point(aes(x = dataset$Level, y = dataset$Salary), colour = 'red')+
  geom_line(aes(x = dataset$Level, y = predict(lin_reg, newdata = dataset)), colour = 'blue')+
  geom_line(aes(x = dataset$Level, y = predict(poly_reg, newdata = dataset)), colour = 'black')+
  xlab('Level')+
  ylab('Salary')

#predicting a new result with linear regression
y_pred = predict(lin_reg,data.frame(Level = 6.5)) 
                 
#predicting a new result with polynomial regression
y_polypred = predict(poly_reg,data.frame(Level = 6.5,
                                     level2 = 6.5^2,
                                     level3 = 6.5^3,
                                     level4 = 6.5^4))
